lib/cocoapods/installer/analyzer.rb

```diff
@@ -636,7 +636,7 @@ def sources
       def verify_platforms_specified!
         unless config.integrate_targets?
           podfile.target_definition_list.each do |target_definition|
-            unless target_definition.platform
+            if target_definition.dependencies.size > 0 && target_definition.platform.nil?
```

Comment:
@segiddins is it correct to exclude inherited dependencies in the check?
If have a `Podfile` like:
```ruby
platform :ios
pod 'InheritedPod'
target 'osx' do
  # empty
end
```
should the `osx` target fail the platform validation?